featurebyte/service/catalog.py

```diff
@@ -22,51 +15,12 @@ class CatalogService(BaseDocumentService[CatalogModel, CatalogCreate, CatalogSer
 
     document_class = CatalogModel
 
-    async def _ensure_default_catalog_available(self) -> None:
-        """
-        Ensure default catalog exists, and create it if it doesn't
-        """
-        try:
-            await super().get_document(document_id=DEFAULT_CATALOG_ID)
-        except DocumentNotFoundError:
-            await super().create_document(
-                CatalogCreate(_id=DEFAULT_CATALOG_ID, default_feature_store_ids=[], name="default")
-            )
-
-    async def create_document(self, data: CatalogCreate) -> CatalogModel:
-        await self._ensure_default_catalog_available()
-        return await super().create_document(data)
 
-    async def get_document(
-        self,
-        document_id: ObjectId,
-        exception_detail: str | None = None,
-        use_raw_query_filter: bool = False,
-        **kwargs: Any,
-    ) -> CatalogModel:
-        await self._ensure_default_catalog_available()
-        return await super().get_document(
-            document_id=document_id,
-            exception_detail=exception_detail,
-            use_raw_query_filter=use_raw_query_filter,
-            **kwargs,
-        )
+class AllCatalogService(CatalogService):
+    """
+    AllCatalogService class
+    """
 
-    async def list_documents_as_dict(
-        self,
-        page: int = 1,
-        page_size: int = DEFAULT_PAGE_SIZE,
-        sort_by: str | None = "created_at",
-        sort_dir: SortDir = "desc",
-        use_raw_query_filter: bool = False,
-        **kwargs: Any,
-    ) -> dict[str, Any]:
-        await self._ensure_default_catalog_available()
-        return await super().list_documents_as_dict(
-            page=page,
-            page_size=page_size,
-            sort_by=sort_by,
-            sort_dir=sort_dir,
-            use_raw_query_filter=use_raw_query_filter,
-            **kwargs,
-        )
+    @property
+    def is_catalog_specific(self) -> bool:
```

Comment:
might be a bit overkill for now, but can consider extracting into a mixin and injecting into each of  these services that are `All*`

Reply:
Any reason why we want to block catalog-specific services from using more general services?

Reply:
Oh sorry, I meant it the other way around. All* accessing catalog specific ones.

Reply:
Catalog specific services requires a valid catalog_id for initialization, so they can only be initialized in a request with `active-catalog-id` specified. If these services can be initialized in the lazy app container then they will respect the specified catalog scope even if used in more generic services